Sources/TuistKit/Services/DocService.swift

```diff
@@ -0,0 +1,122 @@
+}
+
+// MARK: - DocService
+
+struct DocService {
+    private let projectGenerator: ProjectGenerating
+    private let swiftDocController: SwiftDocControlling
+    private let swiftDocServer: SwiftDocServing
+    private let fileHandler: FileHandling
+
+    init(projectGenerator: ProjectGenerating = ProjectGenerator(),
+         swiftDocController: SwiftDocControlling = SwiftDocController(),
+         swiftDocServer: SwiftDocServing = SwiftDocServer(),
+         fileHandler: FileHandling = FileHandler())
+    {
+        self.projectGenerator = projectGenerator
+        self.swiftDocController = swiftDocController
+        self.swiftDocServer = swiftDocServer
+        self.fileHandler = fileHandler
+    }
+
+    func run(project path: AbsolutePath, target targetName: String, serve: Bool, port: UInt16) throws {
+        let (_, graph, _) = try projectGenerator.loadProject(path: path)
+
+        let sources = graph.targets(at: path)
+            .filter { !$0.dependsOnXCTest }
+            .flatMap { $0.target.sources }
+            .map { $0.path }
+
+        try withTemporaryDirectory { generationDirectory in
+            let parameters = Parameters(serve: serve,
+                                        swiftDocServer: swiftDocServer,
+                                        port: port,
+                                        generationDirectory: generationDirectory.pathString)
+
+            try swiftDocController.generate(
+                format: parameters.format,
+                moduleName: targetName,
+                baseURL: parameters.baseURL,
+                outputDirectory: generationDirectory.pathString,
+                sourcesPaths: sources
+            )
+
+            let indexPath = generationDirectory.appending(component: parameters.indexName)
+
+            guard fileHandler.exists(indexPath) else {
+                throw Error.documentationNotGenerated
+            }
+
+            if serve {
+                try swiftDocServer.serve(path: generationDirectory, port: port)
+            } else {
+                logger.pretty("You can find the documentation at \(generationDirectory.pathString)")
+            }
+        }
+    }
+}
+
+// MARK: - Parameters
+
+extension DocService {
+    struct Parameters {
+        let format: SwiftDocFormat
+        let indexName: String
+        let baseURL: String
+
+        init(serve: Bool,
+             swiftDocServer: SwiftDocServing,
+             port: UInt16,
+             generationDirectory: String)
+        {
+            if serve {
+                format = .html
+                indexName = "index.html"
+                baseURL = swiftDocServer.baseURL.appending(":\(port)")
+            } else {
+                format = .commonmark
+                indexName = "Home.md"
+                baseURL = generationDirectory
+            }
+        }
+    }
+}
+
+// MARK: - Error
+
+extension DocService {
+    enum Error: FatalError {
+        case targetNotFound(name: String)
```

Comment:
is this error being used anywhere?

Reply:
Nice catch!! I was using it, but it does not make sense now.

Reply:
I think I stopped using it for some reason, but it's being used again now 💪